src/fastoad/cmd/calc_runner.py

```diff
@@ -0,0 +1,178 @@
+
+from fastoad._utils.files import as_path, make_parent_dir
+from fastoad.io import DataFile
+from fastoad.io.configuration import FASTOADProblemConfigurator
+from fastoad.openmdao.variables import VariableList
+
+# Import MPI4Py at the module level to avoid repeated imports
+try:
+    from mpi4py import MPI
+    from mpi4py.futures import MPIPoolExecutor
+except ImportError:
+    HAVE_MPI = False
+else:
+    HAVE_MPI = True
+
+
+_LOGGER = logging.getLogger(__name__)  # Logger for this module
+
+
+@dataclass
+class CalcRunner:
+    """
+    Class for running FAST-OAD computations for a specific configuration.
+
+    It is specifically designed to run several computations concurrently with
+    :meth:`run_cases`.
+    For each computation, data can be isolated in a specific folder.
+    """
+
+    #: Configuration file, common to all computations
+    configuration_file_path: Union[str, PathLike]
+
+    #: Input file for the computation (will supersede the input file setting in
+    #  configuration file)
+    input_file_path: Optional[Union[str, PathLike]] = None
+
+    #: For activating MDO instead MDA
+    optimize: bool = False
+
+    def run(
+        self,
+        input_values: Optional[VariableList] = None,
+        calculation_folder: Optional[Union[str, PathLike]] = None,
+    ) -> DataFile:
+        """
+        Run the computation.
+
+        This method is useful to set input values on-the-fly, and/or isolate the
+        computation data in a dedicated folder.
+
+        :param input_values: if provided, these values will supersede the content
+                             of input file (specified in configuration file)
+        :param calculation_folder: if specified, all data, including configuration file,
+                                   will be stored in that folder
+        :return: the written data
+        """
+        configuration = FASTOADProblemConfigurator(self.configuration_file_path)
+        if self.input_file_path:
+            configuration.input_file_path = self.input_file_path
+        if calculation_folder:
+            make_parent_dir(calculation_folder)
+            configuration.make_local(calculation_folder)
+
+        problem = configuration.get_problem(read_inputs=True)
+        problem.comm = FakeComm()
+        problem.setup()
+
+        if input_values:
+            for input_variable in input_values:
+                problem.set_val(
+                    input_variable.name,
+                    val=input_variable.val,
+                    units=input_variable.units,
+                )
+
+        if self.optimize:
+            problem.run_driver()
+        else:
+            problem.run_model()
+
+        output_data = problem.write_outputs()
+
+        return output_data
+
+    def run_cases(
+        self,
+        input_list: List[VariableList],
+        destination_folder: Union[str, PathLike],
+        *,
+        max_workers: Optional[int] = None,
+        use_MPI_if_available: bool = True,
```

Comment:
Actually, yes. By using `MPIPoolExecutor`, we require the MPI-2 standard, which is not guaranteed by the presence of an MPI implementation, and, AFAIK, cannot be directly check (I think the `MPIPoolExecutor` class will be provided by `mpi4py` whatever the available MPI implementation).
Also, it might be a corner case, but I don't want a user to be stuck if they have a faulty MPI implementation they can't get rid of (lack of admin privileges, for instance).

Reply:
Got it, thanks for the explanations.